contracts/bridge/BridgeFee.sol

```diff
@@ -19,16 +19,36 @@ contract BridgeFee {
         receiver = _receiver;
     }
 
-    function _payKLAYFee(uint256 _fee) internal {
-        if(isValidReceiver() == true && _fee > 0) {
-            receiver.transfer(_fee);
+    function _payKLAYFeeAndRefundChange(uint256 _feeLimit) internal returns(uint256) {
+        uint256 fee = feeOfKLAY;
+
+        if(isValidReceiver() == true && fee > 0) {
+            require(_feeLimit >= fee, "insufficient feeLimit");
+
+            receiver.transfer(fee);
+            msg.sender.transfer(_feeLimit.sub(fee));
+
+            return fee;
         }
+
+        msg.sender.transfer(_feeLimit);
+        return 0;
     }
 
-    function _payERC20Fee(address _token, uint256 _fee) internal {
-        if(isValidReceiver() == true && _fee > 0) {
-            IERC20(_token).transfer(receiver, _fee);
+    function _payERC20FeeAndRefundChange(address from, address _token, uint256 _feeLimit) internal returns(uint256){
+        uint256 fee = feeOfERC20[_token];
+
+        if (isValidReceiver() == true && fee > 0) {
```

Comment:
What happens if the receiver is invalid? It just returns the feeLimit back to the sender?
What is the meaning of the invalid receiver since the receiver should be set in the constructor.?

Reply:
if there is no feeReceiver, there is no fee. So feeLimit will be refund. 
For now, if isValidReceiver is false, feeReceiver is not set or set zero address.

Reply:
What is the meaning of "invalid"? only "0" is invalid? What about that case that the user set the wrong fee receiver address? Do we have a way to check "validness" of the address?

It seems that the fee receiver address is set when the constructor is called. This means there is no chance that the fee receiver address is not set. In that case, do we really need to check the "validness" of the address of the fee receiver?

Reply:
For now, via BridgeFee constructor, Bridge's constructor set the `address(0)`.
After that, user (SP) can set feeReceiver.

How about change code like below?=)
1. `isValidReceiver() == true` ->`feeReceiver != address(0)`

Reply:
As we discussed, the change is up to you. I just wanted to know the logic and the meaning of "validness". :)